homeassistant/components/blebox/__init__.py

```diff
@@ -36,12 +42,16 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up BleBox devices from a config entry."""
-    websession = async_get_clientsession(hass)
-
     host = entry.data[CONF_HOST]
     port = entry.data[CONF_PORT]
+
+    username = get_non_empty_key(entry.data, CONF_USERNAME)
+    password = get_non_empty_key(entry.data, CONF_PASSWORD)
+
     timeout = DEFAULT_SETUP_TIMEOUT
 
+    websession = get_maybe_authenticated_session(hass, password, username)
+
     api_host = ApiHost(host, port, timeout, websession, hass.loop)
```

Comment:
Why are you passing the username and password in the config flow to `ApiHost` and not here?

Reply:
Oh, my oversight.